oonib/testhelpers/http_helpers.py

```diff
@@ -390,6 +394,9 @@ def tcp_connect(self, socket):
         except ConnectionRefusedError:
             socket_info['status'] = False
             socket_info['failure'] = 'connection_refused_error'
+        except ConnectError:
+            socket_info['status'] = False
+            socket_info['failure'] = 'connect_error'
```

Comment:
Reading what you wrote in TheTorProject/ooni-probe#457, I understand that you have seen the "no route to host" error. Perhaps, it would be interesting if such network error could become the error message as opposed to the more general "connect_error". Saying this because "no route to host" tells actually some precise information on how the host is censored, while "connect_error" strikes me as too broad. Is there a way (and/or a spec) to add information to the "failure" string? Meaning: I am fine with "connect_error" as the first cut error, but I'd rather also see the exact error afterwards.